src/interface/obsidian/src/chat_view.ts

```diff
@@ -962,7 +962,9 @@ export class KhojChatView extends KhojPaneView {
             let endpoint = `${this.setting.khojUrl}/api/chat/sessions`;
             if (agentSlug) {
                 endpoint += `?agent_slug=${encodeURIComponent(agentSlug)}`;
-            }
+            } else if (this.currentAgent)  {
+				endpoint += `?agent_slug=${encodeURIComponent(this.currentAgent)}`;
+			}
```

Comment:
Let's fallback to set agentSlug to this.currentAgent here? This will automatically handle setting the agent in the agent dropdown when the chat is first created.

Currently it is unset due to `this.currentAgent = agentSlug || null` and `agentSelect.value = this.currentAgent || ""` logic later


```suggestion
            let endpoint = `${this.setting.khojUrl}/api/chat/sessions`;
            agentSlug = agentSlug || this.currentAgent;
            if (agentSlug) {
                endpoint += `?agent_slug=${encodeURIComponent(agentSlug)}`;
            }
```

Reply:
Fixed